tests/unit/lms/views/exceptions_test.py

```diff
@@ -1,112 +1,138 @@
 
-    expected_result = None
-    response_status = None
+    @pytest.fixture
+    def exception(self):
+        return httpexceptions.HTTPNotFound()
 
-    def handle(self, pyramid_request):
-        if self.exception is None:
-            return type(self).view(  # pylint:disable=not-callable
-                mock.sentinel.exception, pyramid_request
-            )
 
-        return type(self).view(  # pylint:disable=not-callable
-            self.exception, pyramid_request
+class TestForbidden:
+    def test_it(self, assert_response, exception_views):
+        assert_response(
+            exception_views.forbidden(),
+            403,
+            message="You're not authorized to view this page",
         )
 
-    def test_it_sets_response_status(self, pyramid_request):
-        self.handle(pyramid_request)
+    @pytest.fixture
+    def exception(self):
+        return httpexceptions.HTTPForbidden()
 
-        assert pyramid_request.response.status_int == self.response_status
 
-    def test_it_produces_the_expected_template_vars(self, pyramid_request):
-        result = self.handle(pyramid_request)
+class TestHTTPClientError:
+    def test_it(self, assert_response, exception, exception_views):
+        assert_response(
+            exception_views.http_client_error(),
+            status=exception.status_int,
+            message="http_client_error_test_explanation",
+        )
 
-        assert result == self.expected_result
+    @pytest.fixture
+    def exception(self):
+        return httpexceptions.HTTPBadRequest("http_client_error_test_explanation")
 
 
-class TestNotFound(ExceptionViewTest):
-    view = exceptions.notfound
-    exception = None
+class TestHAPIError:
+    def test_it(self, assert_response, exception_views):
+        assert_response(
+            exception_views.hapi_error(), 500, message="hapi_error_test_explanation"
+        )
 
-    response_status = 404
-    expected_result = {"message": "Page not found"}
+    @pytest.fixture
+    def exception(self):
+        return HAPIError("hapi_error_test_explanation")
 
 
-class TestForbidden(ExceptionViewTest):
-    view = exceptions.forbidden
-    exception = None
+class TestValidationError:
+    def test_it(self, assert_response, exception, exception_views):
+        assert_response(
+            exception_views.validation_error(),
+            status=exception.status_int,
+            error=exception,
+        )
 
-    response_status = 403
-    expected_result = {"message": "You're not authorized to view this page"}
+    @pytest.fixture
+    def exception(self):
+        return ValidationError(sentinel.messages)
 
 
-class TestHTTPClientError(ExceptionViewTest):
-    view = exceptions.http_client_error
-    exception = HTTPBadRequest("This is the error message")
+class TestReusedToolGUIDError:
+    def test_it(self, assert_response, exception_views, pyramid_request):
+        assert_response(exception_views.reused_tool_guid_error(), 200)
```

Comment:
This is apparently a bug in the existing behaviour: when `ReusedToolGUIDError` is raised our error page has a `200 OK` response status. It should probably be a `400`. The existing tests for `reused_tool_guid_error()` on master just don't test the status code. For this PR I'm just moving the test methods into a class, not changing them. But I'll send a follow-up PR to correct this

Reply:
Fix: https://github.com/hypothesis/lms/pull/3166